server/services/v1/query_lifecycle.go

```diff
@@ -71,6 +76,8 @@ func (q *queryLifecycle) run(ctx context.Context, options *ReqOptions) (*Respons
 		log.Debug().Str("ns", tenant.String()).Msg("tenant found")
 	}
 
+	ctx = q.cdcMgr.WrapContext(ctx)
```

Comment:
do we need it here if we just have it in the runner?

Reply:
we need it here because the `tx.Commit(ctx)` call in the lifecycle needs to have the wrapped context